collection-scripts/version

```diff
@@ -1,17 +1,9 @@
 #!/usr/bin/env bash
 
 function version() {
-  # get version from image
-  version=$( \
-    oc status | grep '^pod' | \
-    sed -n -r -e 's/.*([[:digit:]]+\.[[:digit:]]+(:?\.[[:digit:]])?(:?-[^@]+)?).*/\1/p' \
-  )
-
-  # if version not found, fallback to imageID
-  [ -z "${version}" ] && version=$(oc status | grep '^pod.*runs' | sed -r -e 's/^pod.*runs //')
-
-  # if version still not found, use Unknown
-  [ -z "${version}" ] && version="Unknown"
-
-  echo ${version}
+  if [[ ! -z $OS_GIT_VERSION ]] ; then
```

Comment:
`OS_GIT_VERSION` is defined only during the building phase. Once built, the env gets unpopulated. What about to inject the env value in https://github.com/openshift/must-gather/blob/master/Dockerfile.rhel7? E.g. re-generating the `version` file from scratch?

Reply:
The following are embedded into the images during build, and as such are available when the image is run:
```
'OS_GIT_MAJOR': major_version,
'OS_GIT_MINOR': minor_version,
'OS_GIT_PATCH': patch_version,
'OS_GIT_VERSION': f'{major_version}.{minor_version}.{patch_version}{release_suffix}',
'OS_GIT_TREE_STATE': 'clean',
'SOURCE_GIT_TREE_STATE': 'clean',
'BUILD_VERSION': version,
'BUILD_RELEASE': release if release else '',
```
You can download the [must-gather.tar](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_must-gather/327/pull-ci-openshift-must-gather-master-e2e-aws/1597782051321286656/artifacts/e2e-aws/gather-must-gather/artifacts/must-gather.tar) from one of the jobs on this PR, and see that the `version` file contains:
```
openshift/must-gather
4.13.0-202211071433.p0.g63e9a04.assembly.stream-63e9a04
```

Reply:
@ingvagabund I've updated the enhancement to specify the SemVer format (original intention) used elsewhere in OpenShift (vs. the OSGi-type format specified there originally).

Reply:
With https://github.com/openshift/enhancements/pull/1294 merged we are good here. @soltysh would you please approve the PR?